plugins/bkpr/bookkeeper.c

```diff
@@ -386,6 +395,16 @@ static struct command_result *json_list_account_events(struct command *cmd,
 		channel_events = account_get_channel_events(cmd, db, acct);
 		chain_events = account_get_chain_events(cmd, db, acct);
 		onchain_fees = account_get_chain_fees(cmd, db, acct);
+	} else if (payment_id != NULL) {
+		channel_events = get_channel_events_by_id(cmd, db, payment_id);
+
+		// Transaction ids are stored as big-endian in the database
+		reverse_bytes(payment_id->u.u8, sizeof(payment_id->u.u8));
+		tx_id = tal(cmd, struct bitcoin_txid);
+		tx_id->shad.sha = *payment_id;
```

Comment:
nit: put the payment_id data into the tx_id, and then reverse the bytes. this leaves the `payment_id` data untouched/true to self.